presto-main/src/main/java/io/prestosql/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1450,7 +1455,7 @@ else if (node.getType() == FULL) {
                 Expression expression = ((JoinOn) criteria).getExpression();
 
                 // need to register coercions in case when join criteria requires coercion (e.g. join on char(1) = char(2))
-                ExpressionAnalysis expressionAnalysis = analyzeExpression(expression, output);
+                ExpressionAnalysis expressionAnalysis = analyzeExpression(expression, output, node.getType() == INNER ? Correlation.ALLOWED : Correlation.DISALLOWED);
```

Comment:
Is it `DISALLOWED` because planner cannot handle them or ANSI SQL disallows such queries? I think we need to distinguish these two situations with different error messages.

Reply:
It’s disallowed from a planner/analyzer perspective. I’m not aware of any case that’s disallowed by the spec